src/cpp/cpp_typecheck_virtual_table.cpp

```diff
@@ -71,7 +71,7 @@ void cpp_typecheckt::do_virtual_table(const symbolt &symbol)
       id2string(vt_symb_type.name) + "@"+ id2string(symbol.name);
     vt_symb_var.base_name=
       id2string(vt_symb_type.base_name) + "@" + id2string(symbol.base_name);
-    vt_symb_var.mode=ID_cpp;
```

Comment:
Unrelated change?

Reply:
Hmm, yes, I'll remove it.

Reply:
Hmm, looking at my commit message that's actually explained:

    Do not shadow "mode" in cpp front-end

    Also pick up a previously set mode from a symbol rather than hard-coding ID_cpp
    where possible. The symbol's mode could be ID_C or ID_cpp, depending on the
    linkage specification.

Reply:
I'm a little uncomfortable with either the commit message, or the commit - the headline of the message 'Do not shadow "mode" in cpp front-end' sounds like a simple refactoring, yet the commit *also* has a functional change in there too (which admittedly is mentioned in the body of the message) - i.e. the handling of previously set modes. Might it be better to either split that change out in to a separate commit, or change the wording of this commit so that the headline is more along the lines of 'Improve handling of "mode" in cpp frontend' ?

Reply:
I have taken the route of amending the commit message as suggested.